### ISSUE TITLE
Discovery: pretrained open tabular foundation model as a general local typed decision engine

### ISSUE THREAD

**Issue by Unjuno** [Unjuno/agent-interface#853]
TASK: `LOCAL-SYSTEM1-TABULAR-FM-FEASIBILITY-20260917-001`

Publication/intake BASE: `824fe51dc35da56782a0a1817889bff22b9b8b64`. Parent: #9. Related: #824/#830/#832/#834/#838, #751, #745/#755, #12/#15.

## Why this is a genuinely new rung

The completed sparse-policy ladder #824→#830→#832→#834→#838 established a useful negative boundary: task-local KNN/linear/tree/RF policies are cheap to fit and extremely cheap to execute, and uncertainty-guided teacher acquisition raises raw fidelity substantially, but the exact safe-abstention gate was not robust at 256 synthetic labels. #838 explicitly stops threshold/tree/query tuning and blocks escalation to real Astra/GUI action from that evidence.

This issue does **not** continue that learner-tuning lane. It tests a different architecture suggested by the Jev comparison and current open tabular-foundation-model literature: ship one pretrained general local in-context learner and specialize it to the current typed decision problem by supplying a small support set, with **no task-specific gradient update**.

Jev remains an external inspiration/comparator only; it is not a dependency, implementation source or required service.

A concrete first candidate is **Mitra-v2 classifier** (`autogluon/mitra-classifier-2`): a 75.7M-parameter tabular foundation model trained on synthetic tasks, released with weights/inference code under Apache-2.0. The candidate is replaceable; the architectural question is whether a permissively licensed pretrained in-context learner can serve as Agent Interface's weak general local System-1 over structured state.

## Target architecture

```text
Astra / frontier planner
    defines typed decision contract
    supplies/corrects sparse support examples when useful
             ↓
General Local System-1
    pretrained once
    task adaptation by in-context support, no gradient update
    structured state -> typed scores/probabilities/YIELD
             ↓
Decision gate / evidence-role checks
             ↓
Deterministic authority runtime
    lease/focus/surface/allowed-action/release
             ↓
Computer
```

Hard invariants:

```text
prediction != authority
confidence != correctness evidence
historical/support example != current admission evidence
UNKNOWN/YIELD must remain representable
```

The local model may rank/choose only within the currently declared candidate set. It cannot grant/extend authority, refresh stale evidence, invent a new semantic goal, or suppress release/cancel.

## Rung 0 — CPU high-cadence feasibility before efficacy

Do **not** spend Astra/model labels or run a GUI/game first. The first discriminator is whether a general pretrained in-context learner is even in the right latency/memory class for local control.

### H

With a frozen 8-feature / 6-class typed decision shape and 256 labelled support rows, a pretrained Mitra-v2 classifier running fully locally with no network and no gradient update can reuse one installed support context and produce repeated single-state typed decisions fast enough to be operationally distinct from a frontier-model boundary. Cold load/context setup is accounted separately from warm per-decision inference.

### T

Fresh disposable container only. No repository/shared-runtime/GUI/game/model-provider/user-data action.

Before formal execution:

1. pin exact Mitra-v2 model repository revision, checkpoint SHA-256, inference-code/package version and Apache-2.0 license bytes;
2. record CPU model, logical/physical cores, RAM, PyTorch/runtime versions and thread settings;
3. generate one deterministic numeric support/query fixture with shape `support=(256,8)`, `classes=6`, plus 1,024 frozen query rows; labels need only exercise all six classes because Rung 0 is timing/ABI, not accuracy;
4. freeze runner/auditor/fixture/model identities before timing;
5. prohibit network access in the measured process after weights are materialized.

Measure separately:

- process/model cold-load wall time;
- support/context setup or `fit` wall time (ICL setup, not weight training);
- warm single-query latency distribution after explicit warm-up;
- batch throughput as secondary diagnostic only;
- peak RSS;
- returned class/probability shape, finite-value checks, and explicit mapping to the six declared candidate IDs;
- repeated-query determinism under the frozen seed/settings.

Primary cadence dispositions use warm single-query p95:

```text
REALTIME_10HZ    p95 < 100 ms/state
INTERACTIVE_2HZ  100 ms <= p95 < 500 ms/state
BATCH_ONLY       p95 >= 500 ms/state
```

These are capability classes, not accuracy claims. Also retain p50/p99 and exact hardware; do not compare numbers across hardware without qualification.

### D

`PASS_GENERAL_LOCAL_SYSTEM1_RUNTIME_SCOPED` only if:

- pinned weights/code/license integrity passes;
- no task-specific gradient update occurs;
- all 1,024 warm outputs map to the frozen six-class typed decision vocabulary with finite scores/probabilities;
- no timed-region network access occurs;
- peak RSS and all timing endpoints are retained;
- warm single-query p95 is `<500 ms`, establishing at least an interactive local decision class.

If p95 `<100 ms`, additionally record `REALTIME_10HZ_CANDIDATE`; this authorizes a separately frozen efficacy/safe-abstention rung, not action authority.

`REJECT_CPU_HIGH_CADENCE_SHAPE` if warm p95 is `>=500 ms` or memory/runtime requirements are incompatible with an ordinary CPU-only installation. A GPU successor, quantized/distilled model, or smaller open TFM would then be a **new** hypothesis; do not relabel the CPU result.

`FAIL_INTEGRITY` for weight/source/license mismatch, network use in the timed region, hidden fine-tuning, or timing that mixes cold setup with warm decisions.

## Rung 1 — only if Rung 0 is viable

A separately frozen experiment will compare the surviving pretrained ICL backend against the smallest conventional local baseline on a fresh matched typed-decision task. The intervention is learner architecture only; support rows, state representation, class vocabulary, held-out ordinary/stress rows and confidence/YIELD policy are identical.

Required outputs include raw fidelity, safe abstention, teacher-YIELD→executable errors, context/setup cost, warm latency, and break-even reuse count. The #838 first outcome remains negative evidence and is not pooled, re-thresholded or reinterpreted.

No real Astra labels or real input are authorized until a new representation/model passes the independently frozen shadow-mode gate.

## C — competing explanations

- Tabular benchmark strength may not transfer to action-policy decision boundaries.
- A 75.7M foundation model may be accurate but too slow for high-cadence CPU control.
- ICL probabilities may still be badly calibrated near YIELD/action boundaries.
- Apparent fast repeated inference may depend on caching that is invalidated whenever support/context changes.
- Numeric structured state may omit the hard perceptual/temporal variables that dominate real GUI/game control.
- A tiny deterministic/linear/tree backend may remain superior after the task has enough verified labels, even if a general TFM is better initially.

## U — scope and uncertainty

Rung 0 is only a local runtime-class test, not DOOM success, 3D manipulation, general intelligence, model quality, or end-to-end Agent Interface speed. Rung 1, if reached, remains a structured-state shadow-policy test. Visual grounding, temporal state extraction, deterministic authority/release and independent effect verification remain separate mechanisms.

The intended product direction is **not** to standardize on Mitra-v2 today. It is to test whether an open pretrained tabular in-context learner is a credible implementation class for a general local typed decision plane, while keeping the Decision ABI/backend replaceable.

**Comment by Unjuno** [comment]
CONSTRUCTION NOTE — formal measurement remains **0 rows**. Disposable `/tmp/ai_exp853` was used only to test whether the current base container can materialize the candidate dependency. A fresh venv could be created, but PyPI resolution repeatedly failed with DNS/name-resolution errors and the package-download command timed out. No Mitra weights were downloaded, no model was loaded, and no timing/accuracy endpoint was inspected.

This is a harness/network provisioning result, **not** evidence against Mitra-v2 or the local-System-1 hypothesis. Do not fold it into Rung0. The next step is to materialize exact package/model bytes through an approved external artifact/compute path, pin revision + SHA-256 + license, then freeze the runner before any scored timing. If that cannot be done reproducibly, record `SETUP_BLOCKED_ARTIFACT_MATERIALIZATION` rather than manufacturing a latency result.

**Comment by Unjuno** [comment]
Coordination update from completed #876: a source-first frozen custom numeric typed-decision kernel (33 heads / 455 logits including one 255-way choice, batch1, 1 thread) kept a deliberately oversized **16,777,216-float32-parameter** installed decision bundle at warm p95 **11.3963 ms/state** on Intel Xeon Platinum 8370C, with explicit YIELD and no training/network/authority path. #876 is runtime-shape evidence only; random parameters provide no intelligence/calibration claim.

This does **not** invalidate #853. It narrows its value proposition: a 75.7M-class pretrained TFM is no longer needed to answer whether typed local decisions can be fast. #853 should now justify itself only as a **representation/generalization/calibration backend** that materially outperforms the smallest compiled rule/linear/tree alternative on a real matched decision problem. Preserve its current formal count0 and artifact-materialization note unchanged; do not reinterpret the setup block or #824/#838 fidelity negatives.

**Comment by Unjuno** [comment]
## Fresh local feasibility gate — 2026-09-21

This is an environment check only; it does not alter the existing artifact-materialization note and does not start Rung 0.

- **H/T/D/C/U:** H: the prescribed CPU ICL timing rung cannot be allocated while the machine's system volume is full and no disposable container engine is running. T: read-only check of local storage/runtime before any package/model materialization. D: **STOP_SETUP_LOCAL_CAPACITY**; formal rows remain 0. C: Windows host reports C: free bytes `0`, `com.docker.service` stopped, Docker Desktop Linux engine pipe absent; detected GPU is RTX 3080 Laptop (the specified Rung 0 is CPU, so GPU is not being used as a substitute). No package download, model load, network/model call, storage cleanup, service restart, Docker repair, or formal timing was attempted. U: this is not a Mitra performance/quality result and does not replace the previously recorded PyPI DNS/materialization failure.

Next valid step remains restoring safe local disk headroom and making the disposable container/runtime available, then pinning and materializing exact package, weight and license bytes before freezing any timing run. Preserve this STOP separately; do not retry or fold it into the formal measurement.
